source/io/module_eggShell.js

```diff
@@ -423,6 +429,54 @@
             return numericValue;
         };
 
+        Module.eggShell.dataReadInt8Array = function (arrayPointer) {
```

Comment:
The existing EggShell_GetArrayMetadata does not return the length of the array, but you say we should not add Data_GetTypedArrayLength. How else could I get the array length without adding a function like Data_GetTypedArrayLength?

Reply:
As for refactoring the dataReadAsXXArray methods, what would you think if I added additional fields to what memo suggested so for example:

```javascript
ArrayInt16: {
                reader: Module.eggShell.dataReadInt16Array,
                writer: Module.eggShell.dataWriteInt16Array,
                isValidReturnType: isInt16Array,
                heap: supportedArrayTypeConfig.Int16,
                constructor: Int16Array
           }
```

?

Reply:
Sorry about that, I overlooked that GetNumericArray was using the EggShell_GetArrayDimLength function. But following the same pattern as the others, instead of GetTypedArrayLength which hard codes reading dimension 0, make a Data_GetArrayDimLength and an EggShell_GetArrayDimLength.

Reply:
Right now these metadata objects are in different files so I don't think it is worth merging them yet unless we have a good way to describe what is being represented.

Then maybe we would also merge it with the object used by getNumericArray and move it to core or something.